client/server/render/index.js

```diff
@@ -233,12 +234,15 @@ export function serverRender( req, res ) {
 		attachHead( context );
 
 		const cacheableReduxSubtrees = [ 'documentHead' ];
-		const isomorphicSubtrees = context.section?.isomorphic ? [ 'themes', 'ui' ] : [];
+		const isomorphicSubtrees = context.section?.isomorphic
+			? [ 'themes', 'ui', 'productsList' ]
```

Comment:
Adding the`productsList` here avoids the situation where the server fetches the plugin prices in a different currency than the client. It should also improve performance by not having to fetch the `productsList` again on the client side.

Reply:
By the way, you can now add this below on line 258 to improve logged out performance -- assuming that caching `productsList` in the state allows us to avoid an extra network request on the server. We improved state caching so that it doesn't operate for logged-in requests now. 

(Obviously only do this if the data in `productsList` matches 1:1 with the route after accounting for locale and everything!)

Reply:
Thanks @noahtallen for the improvements to the state caching. I added `productsList` and it seems to work well.